### PR TITLE
Bump CAPI to v1.2.0

### PR DESCRIPTION
**Pull request by abhay-krishna**
/hold

Bumping cluster-api to [latest](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.2.0).

#### Testing

Following steps were done to test the new `clusterctl` and CAPI managers:
1. Patches were regenerated by checking out v1.1.3 tag, applying patches, then checking out v1.2.0 tag and cherrypicking commits from the patches on top of v1.2.0 tag.
2. Building/pushing images for all CAPI managers from patched repo.
3. Building/pushing CLI tools image embedded with v1.2.0 clusterctl.
4. Building manifests from v1.2.0 tag (generated into _output directory).
4. Downloading bundle release manifest 
locally and replacing URIs for CAPI controller managers, CLI tools image and CAPI manifests.
5. Validating CAPD cluster create/delete using bundles override.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/kubernetes-sigs/cluster-api/patches/0003-Unstacked-etcd-and-controlplane-upgrade.patch [hunk 1]
```diff
@@ -63,21 +63,21 @@ index 58ef4a74e..1c7d411bd 100644
...
  // MachineAddressType describes a valid MachineAddress type.
 diff --git a/api/v1beta1/common_types.go b/api/v1beta1/common_types.go
-index 7d19c0e05..f17d3b066 100644
+index 1d8d0d633..b1ffbd135 100644
 --- a/api/v1beta1/common_types.go
 +++ b/api/v1beta1/common_types.go
-@@ -112,6 +112,9 @@ const (
- 	// An external controller must fulfill the contract of the InfraCluster resource.
- 	// External infrastructure providers should ensure that the annotation, once set, cannot be removed.
- 	ManagedByAnnotation = "cluster.x-k8s.io/managed-by"
```
Comment (vivek-koppuru) [inline projects/kubernetes-sigs/cluster-api/patches/0003-Unstacked-etcd-and-controlplane-upgrade.patch:72]: We are removing this annotation?

File: projects/kubernetes-sigs/cluster-api/patches/0002-Add-unstacked-etcd-support.patch [hunk 2]
```diff
@@ -539,109 +540,17 @@ index fde450016..943898620 100644
...
-+                            type: array
-+                        type: object
-                     type: object
-                   mounts:
-                     description: Mounts specifies a list of mount points to be setup.
-@@ -2878,6 +2892,20 @@ spec:
-                               upgrades.
-                             type: string
-                         type: object
-+                      proxy:
```
Comment (vivek-koppuru) [inline projects/kubernetes-sigs/cluster-api/patches/0002-Add-unstacked-etcd-support.patch:582]: So we are just moving this to the first patch instead?

File: projects/kubernetes-sigs/cluster-api/patches/0005-Make-pause-and-bottlerocket-bootstrap-images-updatab.patch [hunk 3]
```diff
@@ -31,18 +31,19 @@ index 9d5e58664..c763f6906 100644
...
+ 		{spec, kubeadmConfigSpec, joinConfiguration, patches, directory},
  		{spec, kubeadmConfigSpec, preKubeadmCommands},
- 		{spec, kubeadmConfigSpec, postKubeadmCommands},
 diff --git a/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook_test.go b/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook_test.go
-index 9923c9527..62dbf938e 100644
+index 4d111c6b3..629385c6d 100644
 --- a/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook_test.go
 +++ b/controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook_test.go
-@@ -594,6 +594,18 @@ func TestKubeadmControlPlaneValidateUpdate(t *testing.T) {
- 	validIgnitionConfigurationAfter := validIgnitionConfigurationBefore.DeepCopy()
```
Comment (vivek-koppuru) [inline projects/kubernetes-sigs/cluster-api/patches/0005-Make-pause-and-bottlerocket-bootstrap-images-updatab.patch:44]: This logic changed to the below?

File: projects/kubernetes-sigs/cluster-api/patches/0003-Unstacked-etcd-and-controlplane-upgrade.patch (hunk 1 above)
Comment (abhay-krishna) [reply projects/kubernetes-sigs/cluster-api/patches/0003-Unstacked-etcd-and-controlplane-upgrade.patch:72]: So these are just the context lines for the diff, they're not getting modified by the patch. Between 1.1.3 and 1.2.0, they had added the `TopologyDryRunAnnotation`, that now acts as the context lines, replacing this one. The actual patch is line 73 to 75, where we add the `ControlPlaneUpgradeCompletedAnnotation`.

Comment (vivek-koppuru) [reply projects/kubernetes-sigs/cluster-api/patches/0003-Unstacked-etcd-and-controlplane-upgrade.patch:72]: Oh right I forgot, thanks for that

File: projects/kubernetes-sigs/cluster-api/patches/0005-Make-pause-and-bottlerocket-bootstrap-images-updatab.patch (hunk 3 above)
Comment (vivek-koppuru) [reply projects/kubernetes-sigs/cluster-api/patches/0005-Make-pause-and-bottlerocket-bootstrap-images-updatab.patch:44]: This is probably the context lines too

Comment (abhay-krishna) [reply projects/kubernetes-sigs/cluster-api/patches/0005-Make-pause-and-bottlerocket-bootstrap-images-updatab.patch:44]: Same as above, they added a few lines below this in [this commit](https://github.com/kubernetes-sigs/cluster-api/commit/0146e81d7c51722de3d74d794cc948c75d959aaf), so we got new context lines.

**Review by vivek-koppuru — APPROVED** [review]

File: projects/kubernetes-sigs/cluster-api/patches/0002-Add-unstacked-etcd-support.patch (hunk 2 above)
Comment (abhay-krishna) [reply projects/kubernetes-sigs/cluster-api/patches/0002-Add-unstacked-etcd-support.patch:582]: yeah, for a couple of patches that changed API structs, the `generate-manifests` hadn't been run to update CRDs until the next patch. But this time I ran the generate command as soon as i applied the patch to see if there were any CRD changes. This caused CRD changes to moved from one patch to another, in this case, the previous one.

Comment (abhay-krishna) [reply projects/kubernetes-sigs/cluster-api/patches/0002-Add-unstacked-etcd-support.patch:582]: This is also why I removed patch 11 - adding generated manifests. That change had already gone into a preceding patch.

**Pull request merged**